diff --git a/build.gradle b/build.gradle
--- a/build.gradle
+++ b/build.gradle
@@ -7,1 +7,0 @@
-  id "com.jfrog.bintray" version "1.8.4"
@@ -16,0 +16,1 @@
+def gradleApiVersion = GradleVersion.current().getVersion()
@@ -61,6 +62,7 @@ task setupGradleApiDeps(dependsOn: configurations.gradleApiDeps, type: Sync) {
 
     def connection = GradleConnector.newConnector()
       .useGradleVersion(gradleApiVersion)
+      .useGradleUserHomeDir(gradle.getGradleUserHomeDir())
       .forProjectDirectory(projectDir)
       .connect()
     try {
@@ -167,1 +167,2 @@
+/*
 bintray {
@@ -184,1 +194,3 @@
 }
+*/
