diff --git androidx/compose/compiler/plugins/kotlin/ComposePlugin.kt androidx/compose/compiler/plugins/kotlin/ComposePlugin.kt
index dc070b3..e14c8b0 100644
--- androidx/compose/compiler/plugins/kotlin/ComposePlugin.kt
+++ androidx/compose/compiler/plugins/kotlin/ComposePlugin.kt
@@ -301,6 +301,9 @@ class ComposePluginRegistrar : org.jetbrains.kotlin.compiler.plugin.ComponentReg
 
     companion object {
         fun checkCompilerVersion(configuration: CompilerConfiguration): Boolean {
+            // Android Studio: this version check does not apply since we compile Compose
+            // sources directly against the Kotlin compiler bundled in IntelliJ.
+            return true
             try {
                 val KOTLIN_VERSION_EXPECTATION = "2.0.0-Beta4"
                 KotlinCompilerVersion.getVersion()?.let { version ->
