Subject: [PATCH] For text 'Allow bloom to access your camera and microphone', apply text transformation so that all of this is lowercase.
---
Index: onboardingandauthentication/src/main/java/com/ibm/google/onboardingandauthentication/ui/screen/PermissionScreen.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/onboardingandauthentication/src/main/java/com/ibm/google/onboardingandauthentication/ui/screen/PermissionScreen.kt b/onboardingandauthentication/src/main/java/com/ibm/google/onboardingandauthentication/ui/screen/PermissionScreen.kt
--- a/onboardingandauthentication/src/main/java/com/ibm/google/onboardingandauthentication/ui/screen/PermissionScreen.kt	(revision 83d9644fffce800965f3b282e9da0ac53f289f43)
+++ b/onboardingandauthentication/src/main/java/com/ibm/google/onboardingandauthentication/ui/screen/PermissionScreen.kt	(date 1755858371797)
@@ -24,6 +24,7 @@
 import androidx.compose.ui.layout.ContentScale
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.capitalize
 import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
@@ -32,6 +33,7 @@
 import com.ibm.google.onboardingandauthentication.ui.component.RoundedFilledButton
 import com.ibm.google.onboardingandauthentication.ui.theme.OnBoardingAndAuthenticationTheme
 import com.ibm.google.onboardingandauthentication.utils.Utils.maxWidth
+import java.util.Locale
 
 /**
  * Composable that displays the permission request screen.
@@ -75,7 +77,7 @@
                     modifier = Modifier
                         .fillMaxWidth()
                         .padding(horizontal = 32.dp, vertical = 24.dp),
-                    text = stringResource(R.string.allow_bloom_to_access_your_camera),
+                    text = stringResource(R.string.allow_bloom_to_access_your_camera).lowercase(),
                     textAlign = TextAlign.Center,
                     style = MaterialTheme.typography.headlineMedium,
                     color = MaterialTheme.colorScheme.onSurface
